java/kotlin-extractor2/src/main/kotlin/TrapWriter.kt

```diff
@@ -147,31 +147,29 @@ abstract class TrapWriter(
         return label
     }
 
-    /*
-    OLD: KE1
-        /**
-         * It is not easy to assign keys to local variables, so they get given `*` IDs. However, the
-         * same variable may be referred to from distant places in the IR, so we need a way to find out
-         * which label is used for a given local variable. This information is stored in this mapping.
-         */
-        private val variableLabelMapping: MutableMap<IrVariable, Label<out DbLocalvar>> =
-            mutableMapOf<IrVariable, Label<out DbLocalvar>>()
-        /** This returns the label used for a local variable, creating one if none currently exists. */
-        fun <T> getVariableLabelFor(v: IrVariable): Label<out DbLocalvar> {
-            val maybeLabel = variableLabelMapping.get(v)
-            if (maybeLabel == null) {
-                val label = getFreshIdLabel<DbLocalvar>()
-                variableLabelMapping.put(v, label)
-                return label
-            } else {
-                return maybeLabel
-            }
+    /**
+     * It is not easy to assign keys to local variables, so they get given `*` IDs. However, the
+     * same variable may be referred to from distant places in the IR, so we need a way to find out
+     * which label is used for a given local variable. This information is stored in this mapping.
+     */
+    private val variableLabelMapping: MutableMap<KtProperty, Label<out DbLocalvar>> =
```

Comment:
Could you add something like `// TODO: This should be in a subclass so that DiagnosticTrapWriter doesn't include it, as it is not threadsafe` please?